src/common/helpers/snap-builds.js

```diff
@@ -25,43 +25,110 @@
 // };
 
 export const BuildStatusColours = {
+  BLUE: 'blue',
   GREEN: 'green',
   YELLOW: 'yellow',
   RED: 'red',
   GREY: 'grey'
 };
 
+const BuildAndPubState = {
+  'NEVER_BUILT':     createState('Never built', BuildStatusColours.GREY, false, 8),
+  'BUILDING_SOON':   createState('Building soon', BuildStatusColours.GREY, 'ellipsis', 7),
+  'IN_PROGRESS':     createState('In progress', BuildStatusColours.BLUE, 'spinner', 2),
+  'FAILED_TO_BUILD': createState('Failed to build', BuildStatusColours.RED, 'cross', 1),
+  'WONT_PUBLISH':    createState('Built, won\'t be published', BuildStatusColours.GREEN, 'tick_outlined', 6),
+  'PUBLISHING_SOON':    createState('Built, publishing soon', BuildStatusColours.GREY, 'tick', 2),
+  'PUBLISHING_NOW':     createState('Built, publishing now', BuildStatusColours.BLUE, 'tick', 3),
+  'PUBLISHING_FAILED':  createState('Built, failed to publish', BuildStatusColours.RED, 'tick', 4),
+  'PUBLISHED':       createState('Built and published', BuildStatusColours.GREEN, 'tick_filled', 5)
+};
+
+function createState(message, colour, icon, priority) {
+  return {
+    message,
+    colour,
+    icon,
+    priority
+  };
+}
+
 // Based on BuildStatusConstants from LP API
 // https://git.launchpad.net/launchpad/tree/lib/lp/buildmaster/enums.py#n22
 //
-// mapping between build status from LP and red, yellow, green, grey status colours
-const BuildStatusMapping = {
-  'Needs building': BuildStatusColours.YELLOW,
-  'Successfully built': BuildStatusColours.GREEN,
-  'Failed to build': BuildStatusColours.RED,
-  'Dependency wait': BuildStatusColours.RED,
-  'Chroot problem': BuildStatusColours.RED,
-  'Build for superseded Source': BuildStatusColours.RED,
-  'Currently building': BuildStatusColours.YELLOW,
-  'Failed to upload': BuildStatusColours.RED,
-  'Uploading build': BuildStatusColours.YELLOW,
-  'Cancelling build': BuildStatusColours.RED,
-  'Cancelled build': BuildStatusColours.RED
+const LaunchpadBuildStates = {
+  'NEEDSBUILT': 'Needs building',
+  'FULLYBUILT': 'Successfully built',
+  'FAILEDTOBUILD': 'Failed to build',
+  'MANUALDEPWAIT': 'Dependency wait',
+  'CHROOTWAIT': 'Chroot problem',
+  'SUPERSEDED': 'Build for superseded Source',
```

Comment:
@matthewpaulthomas It's for cases such as "tried to build a particular source package version in a PPA but when the build got round to running it turned out that a newer source package version had already been uploaded".  Not the sort of thing that actually happens for snap builds, so it might as well just be considered a build failure if it does show up for some weird reason.